TradeItIosTicketSDK2/TradeItTransactionsViewController.swift

```diff
@@ -6,36 +6,89 @@ class TradeItTransactionsViewController: TradeItViewController, TradeItTransacti
     private let alertManager = TradeItAlertManager()
     private var transactionsTableViewManager: TradeItTransactionsTableViewManager?
-    
+
     @IBOutlet weak var transactionsTable: UITableView!
-    
+    @IBOutlet var transactionsBackgroundView: UIView!
+
     var linkedBrokerAccount: TradeItLinkedBrokerAccount?
-    
+
     override func viewDidLoad() {
         super.viewDidLoad()
-        
+
         guard let linkedBrokerAccount = self.linkedBrokerAccount else {
             preconditionFailure("TradeItIosTicketSDK ERROR: TradeItTransactionsViewController loaded without setting linkedBrokerAccount.")
         }
-        self.transactionsTableViewManager = TradeItTransactionsTableViewManager(linkedBrokerAccount: linkedBrokerAccount)
+        self.transactionsTableViewManager = TradeItTransactionsTableViewManager(linkedBrokerAccount: linkedBrokerAccount, noResultsBackgroundView: transactionsBackgroundView )
         self.transactionsTableViewManager?.delegate = self
         self.transactionsTableViewManager?.transactionsTable = transactionsTable
         self.loadTransactions()
     }
-    
+
     // MARK: IBAction
-    
-    @IBAction func filterButtonWasTapped(_ sender: Any) {
-        //TODO: https://www.pivotaltracker.com/story/show/148168413
+
+    @IBAction func filterButtonWasTapped(_ sender: UIBarButtonItem) {
+        let alertController = UIAlertController(title: nil, message: nil, preferredStyle: UIAlertControllerStyle.actionSheet)
+
+        let allTransactionsAction = provideTransactionsUIAlertAction(filterType: .ALL_TRANSACTIONS)
+        alertController.addAction(allTransactionsAction)
+
+        if let transactionsPresenter = transactionsTableViewManager?.transactionHistoryResultPresenter {
+            if transactionsPresenter.numberOfTransactions(forFilterType: .TRADES) > 0 {
+                let tradesAction = provideTransactionsUIAlertAction(filterType: .TRADES)
+                alertController.addAction(tradesAction)
+            }
+            if transactionsPresenter.numberOfTransactions(forFilterType: .DIVIDENDS_AND_INTEREST) > 0 {
+                let dividendsAndInterestAction = provideTransactionsUIAlertAction(filterType: .DIVIDENDS_AND_INTEREST)
+                alertController.addAction(dividendsAndInterestAction)
+            }
+            if transactionsPresenter.numberOfTransactions(forFilterType: .TRANSFERS) > 0 {
+                let transfersAction = provideTransactionsUIAlertAction(filterType: .TRANSFERS)
+                alertController.addAction(transfersAction)
+            }
+            if transactionsPresenter.numberOfTransactions(forFilterType: .FEES) > 0 {
+                let feesAction = provideTransactionsUIAlertAction(filterType: .FEES)
+                alertController.addAction(feesAction)
+            }
+            if transactionsPresenter.numberOfTransactions(forFilterType: .OTHER) > 0 {
```

Comment:
We're hitting this call a lot which is iterating over the transactions every time. Is there a nicer way to manage this?

Also might be able to dry this up similar to how we did on the other PR